dotCMS/src/main/java/com/dotmarketing/startup/StartupTasksExecutor.java

```diff
@@ -10,346 +13,207 @@
-					}
-					if(!connection.getAutoCommit()){
-						connection.commit();
-					}
-					connection.close();
-				}
-			} catch (Exception e) {
-				Logger.debug(StartupTasksExecutor.class, "Exception: "
-						+ e.getMessage(), e);
-				throw new DotDataException(
-						"Exception finishing upgrade tasks: " + e.getMessage(),
-						e);
-			} finally {
-
-				CloseUtils.closeQuietly(connection);
-			}
-			Logger.info(this, "Finishing upgrade tasks.");
-		}
-
-	}
+    /**
+     * This will create the db version table if it does not already exist
+     * @return
+     */
+	private boolean insureDbVersionTable() {
+
+        try {
+            currentDbVersion();
+            return true;
+
+        } catch (Exception e) {
+            return createDbVersionTable();
+        }
+
+    }
+    
+    /**
+     * Runs with a separate DB connection
+     * @return
+     */
+    private int currentDbVersion() {
+        try (Connection conn = DbConnectionFactory.getDataSource().getConnection()) {
+            DotConnect db =  new DotConnect().setSQL(SELECT);
+            return  db.loadInt("test");
+
+        } catch (Exception e) {
+            throw new DotRuntimeException(e);
+        }
+    }
+    
+    /**
+     * Runs with a separate DB connection
+     * @return
+     */
+    private boolean createDbVersionTable() {
+
+        try (Connection conn = DbConnectionFactory.getDataSource().getConnection()) {
+            new DotConnect().setSQL(createTableSQL()).loadResult(conn);
+            new DotConnect().setSQL(INSERT).addParam(0).addParam(new Date()).loadResult(conn);
+            return true;
+
+        } catch (Exception e) {
+            Logger.debug(this.getClass(), e.getMessage(),e);
+            throw new DotRuntimeException(e);
+        }
+
+    }
+    
+    
+    public void executeStartUpTasks() throws DotDataException {
+        
+        Logger.debug(this.getClass(), "Running Startup Tasks");
+
+
+        String name = null;
+        try {
+            Logger.info(this, "Running Startup Tasks");
+
+
+            for (Class<?> c : TaskLocatorUtil.getStartupRunAlwaysTaskClasses()) {
+                HibernateUtil.startTransaction();
+                name = c.getCanonicalName();
+                name = name.substring(name.lastIndexOf(".") + 1);
+                if (StartupTask.class.isAssignableFrom(c)) {
+                    StartupTask  task = (StartupTask) c.newInstance();
+                    if (task.forceRun()) {
+                        HibernateUtil.startTransaction();
+                        Logger.info(this, "Running Startup Tasks : " + name);
+                        task.executeUpgrade();
+                    } else {
+                        Logger.info(this, "Not Running Startup Tasks: " + name);
+                    }
+                }
+                HibernateUtil.closeAndCommitTransaction();
+            }
+            Logger.info(this, "Finishing startup tasks.");
+        } catch (Throwable e) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A catch statement should never catch throwable since it includes errors.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5632525)